rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -0,0 +1,142 @@
+import abc
+from typing import Any, Mapping, Union, Type
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.typing import TensorType
+from ray.rllib.utils.framework import try_import_tf
+
+
+tf1, tf, tfv = try_import_tf()
+
+tf1.enable_eager_execution()
+
+
+class RLTrainer:
+    """Base class for rllib algorithm trainers.
+
+    Args:
+        module_class: The (MA)RLModule class to use.
+        module_config: The config for the (MA)RLModule.
+        optimizer_class: The optimizer class to use.
+        optimizer_config: The config for the optimizer.
+
+
+    """
+
+    def __init__(
+        self,
+        module_class: Type[RLModule],
```

Comment:
Is my understanding correct? This can be any RLModule, either single agent or multiagent. You will convert it to multi_agent with `.as_multi_agent()` inside `init_trainer`?

Reply:
correct!

Reply:
Ive updated the docstring to reflect